purchase_supplier_approve/wizard/supplier_approval_wizard.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 RGB Consulting - Domantas Sidorenkovas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SupplierValidationWizard(models.TransientModel):
+    _name = "supplier.approval.wizard"
+    _description = "Supplier Validation Wizard"
+
+    def _default_partner(self):
+        return self.env['res.partner'].browse(self._context.get('active_id'))
+
+    partner_id = fields.Many2one(comodel_name='res.partner',
```

Comment:
Looks like, partner_id is not really needed, right? (you need it only because you want to set its default).
On open wizard, I would prefer to use `def default_get(...)` to set default supplier approval state (you can use _default_partner or _get_partner for partner object)
On save, again, you can still use _get_partner to get the partner and write it.

Note:
* This way, we can avoid using _onchange_partner_id, just to get the default approve state. To use method for it right purpose.
* Code can be shorten a bit

WDYT?